packages/@aws-cdk/aws-lambda/lib/log-retention-provider/index.ts

```diff
@@ -49,9 +49,16 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
         // group for this function should already exist at this stage because we
         // already logged the event but due to the async nature of Lambda logging
         // there could be a race condition. So we also try to create the log group
-        // of this function first.
-        await createLogGroupSafe(`/aws/lambda/${context.functionName}`);
-        await setRetentionPolicy(`/aws/lambda/${context.functionName}`, 1);
+        // of this function first. If multiple LogRetention constructs are present
+        // in the stack, they will try to act on this function's log group at the
+        // same time. This can sometime result in an OperationAbortedException. To
+        // avoid this and because this operation is not critical we catch all errors.
+        try {
+          await createLogGroupSafe(`/aws/lambda/${context.functionName}`);
```

Comment:
What am I missing? Where is the part that sends the response to cloudformation?

Reply:
This is only to catch errors when setting the log retention on the lambda provider's log group of the custom resource not the log we are targeting, it's an internal try/catch. The external one still exists.

 https://github.com/awslabs/aws-cdk/blob/8410d48ac570cd998b14569c5d16e08bac8408cc/packages/%40aws-cdk/aws-lambda/lib/log-retention-provider/index.ts#L66-L70

Reply:
We don't want to fail the whole process if setting the retention policy of 1 day on the provider's log group fails.

Reply:
ok. thanks for the clarification. missed that